cukinia

```diff
@@ -621,6 +622,37 @@ _cukinia_mount()
 	return 0
 }
 
+# _cukinia_i2c: Check if i2c device exists
+# Optional: Check of a device address
+# arg1: the bus number
+# arg2: (optional) the device address
+_cukinia_i2c()
+{
+	local bus_number="$1"
+	local device_address="$2"
+	local ret
+
+	_cukinia_prepare "Checking if i2c-$bus_number exists"
+
+	test -d /sys/bus/i2c/devices/i2c-$bus_number
+	ret=$?
+
+	if [ $ret -eq 1 ]
+	then
+		return $ret
+	fi
+
+	if [ -n "$device_address" ]
+	then
+		device_address=$(echo $device_address | sed 's/0x//g')
+		_cukinia_prepare "Checking if device at 0x$device_address is reachable on i2c-$bus_number"
+
+		i2cdetect -y $bus_number 0x$device_address 0x$device_address | cut -d ':' -f2 | grep $device_address
```

Comment:
we should fail early if i2cdetect (an external requirement) isn't present , see _cukinia_kconf() for a similar check (ikconfig)